File: mvd/mvd/report/beratungsstatistik/beratungsstatistik.js
// Copyright (c) 2016, libracore and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Beratungsstatistik"] = {
    "filters": [
        {
            'fieldname': "from_date",
            'label': __("Von"),
            'fieldtype': "Date",
            'reqd': 1,
            'default': frappe.datetime.year_start()
        },
        {
            'fieldname': "to",
            'label': __("Bis"),
            'fieldtype': "Date",
            'reqd': 1,
            'default': frappe.datetime.now_date()
        },
        {
            'fieldname': "sektion_id",
            'label': __("Sektion"),
            'fieldtype': "Link",
            'reqd': 1,
            'options': 'Sektion'
        }
    ]
};

File: mvd/mvd/report/beratungsstatistik/beratungsstatistik.py
# Copyright (c) 2013, libracore and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
    columns, data = get_columns(), get_data(filters)
    return columns, data

def get_columns():
    return[
        {"label": _("Beratungskategorie"), "fieldname": "beratungskategorie", "fieldtype": "Link", "options": "Beratungskategorie"},
        {"label": _("Ist Unterkategorie"), "fieldname": "unterkategorie", "fieldtype": "Check"},
        {"label": _("Anzahl Geschlossen"), "fieldname": "closed", "fieldtype": "Int"}
    ]

def get_data(filters):
    data = []
    closed_as_main = frappe.db.sql("""
                                        SELECT
                                            COUNT(`name`) AS `qty`,
                                            `beratungskategorie`
                                        FROM `tabBeratung`
                                        WHERE `status` = 'Closed'
                                        AND `geschlossen_am` >= '{from_filter}'
                                        AND `geschlossen_am` <= '{to_filter}'
                                        GROUP BY `beratungskategorie`""".format(from_filter=filters.from_date, to_filter=filters.to), as_dict=True)
    
    for cam in closed_as_main:
        data.append({
            'beratungskategorie': cam.beratungskategorie,
            'closed': cam.qty
        })
    
    return data
